collections/slices.go

```diff
@@ -0,0 +1,27 @@
+/*
+Copyright © 2019, 2020 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package collections
+
+// StringInSlice tests whether the given string is contained in slice of strings
+func StringInSlice(a string, list []string) bool {
```

Comment:
I'm wondering if it's possible to have this function for the common case and not only for strings?

Reply:
Unfortunately it is not that easy in Go. I mean, possible to use `interface{}` or so, but it will be a lot slower than. And this function is used in `auth.go` for each incoming request.

Reply:
Gotcha, thanks